after_sign_hook.js

```diff
@@ -1,46 +1,45 @@
-const fs = require('fs');
-const path = require('path');
-var electron_notarize = require('electron-notarize');
-
-module.exports = async function(params) {
-  // Only notarize the app on Mac OS only.
-  if (params.electronPlatformName !== 'darwin') {
-    return;
-  }
-  console.log('afterSign hook triggered', params);
-
-  if (!process.env.CIRCLE_TAG || process.env.CIRCLE_TAG.length === 0) {
-    console.log('Not on a tag. Skipping notarization');
-    return;
-  }
-
-  // Same appId in electron-builder.
-  let appId = 'com.automattic.wordpress';
-
-  let appPath = params.appOutDir
-    ? path.join(
-        params.appOutDir,
-        `${params.packager.appInfo.productFilename}.app`
-      )
-    : params.artifactPaths[0].replace(new RegExp('.blockmap'), '');
-
-  if (!fs.existsSync(appPath)) {
-    throw new Error(`Cannot find application at: ${appPath}`);
-  }
-
-  console.log(`Notarizing ${appId} found at ${appPath}`);
-
-  try {
-    await electron_notarize.notarize({
-      appBundleId: appId,
-      appPath: appPath,
-      appleId: process.env.NOTARIZATION_ID,
-      appleIdPassword: process.env.NOTARIZATION_PWD,
-      ascProvider: 'AutomatticInc',
-    });
-  } catch (error) {
-    console.error(error);
-  }
-
-  console.log(`Done notarizing ${appId}`);
-};
+#!/usr/bin/env node
+
+const path = require( 'path' );
+const { notarize } = require( 'electron-notarize' );
+
+const APP_ID = 'com.automattic.wordpress';
+const NOTARIZATION_ASC_PROVIDER = 'AutomatticInc';
+const NOTARIZATION_ID = process.env.NOTARIZATION_ID;
+const NOTARIZATION_PWD = process.env.NOTARIZATION_PWD;
+
+const shouldNotarize = process.platform === 'darwin' && !!process.env.CIRCLE_TAG;
+
+function elapsed( start ) {
+	const now = new Date();
+
+	const ms = Math.abs( now.getTime() - start.getTime() );
+	const diff = new Date( ms );
+
+	return `${ diff.getMinutes() } minutes, ${ diff.getSeconds() } seconds`;
+}
+
+module.exports = async function( context ) {
+	if ( ! shouldNotarize ) {
+		return;
+	}
+
+	try {
+		const app = path.join( context.appOutDir, `${ context.packager.appInfo.productFilename }.app` );
+		const appName = path.basename( app );
+
+		const start = new Date();
+		console.log( `  • notarizing ${ appName }...` );
+		await notarize( {
+			appBundleId: APP_ID,
+			appPath: app,
+			appleId: NOTARIZATION_ID,
+			appleIdPassword: NOTARIZATION_PWD,
+			ascProvider: NOTARIZATION_ASC_PROVIDER,
+		} )
+		console.log( `  • done notarizing ${ appName }, took ${ elapsed( start ) }` );
+	} catch ( error ) {
+		console.log( error.message );
+		throw ( error );
+	}
```

Comment:
since all we're doing is throwing the error and printing it then is there any reason to catch it? what happens if we don't? won't it still print the error and throw it, doing almost exactly what it does here, or does it not print out the `error.message` because some higher-up code is catching the error and hiding it?

Reply:
It looks a little nicer when output to console, but agree it's extraneous - removed in dbd8b42.

Reply:
I meant the whole `try/catch` block. Right now it's this…

```js
try {
  something()
} catch (anything) {
  throw (anything);
}
```

which I _think_ is equivalent to this…

```js
something();
```

Reply:
tbh was simply unclear what the idiomatic way to "rethrow" in JavaScript (or whether it should be done at all). Seems there is a subtle [difference in stack trace](https://stackoverflow.com/a/49323141) between a throw and rethrow, but doesn't seem suited to this use case. I think doing away with the try/catch here is fine - 17253be6.